From c01473bf00b73ba1dd72813fbc4c4c5d0f66d339 Mon Sep 17 00:00:00 2001
From: Michael Dickens <michael.dickens@ettus.com>
Date: Thu, 23 May 2019 10:26:00 -0400
Subject: [PATCH] blocks: simple fix for Boost 1.70.0 in socket_pdu

NOTE: There have been multiple fixes proposed, such as
https://github.com/gnuradio/gnuradio/pull/2451 .
This one is the simplest and most compatible.
---
 gr-blocks/lib/socket_pdu_impl.cc | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git gr-blocks/lib/socket_pdu_impl.cc gr-blocks/lib/socket_pdu_impl.cc
index e20f1478f0..168d74ebb2 100644
--- gr-blocks/lib/socket_pdu_impl.cc
+++ gr-blocks/lib/socket_pdu_impl.cc
@@ -1,6 +1,6 @@
 /* -*- c++ -*- */
 /*
- * Copyright 2013 Free Software Foundation, Inc.
+ * Copyright 2013,2019 Free Software Foundation, Inc.
  *
  * This file is part of GNU Radio
  *
@@ -165,7 +165,11 @@ namespace gr {
     void
     socket_pdu_impl::start_tcp_accept()
     {
+#if (BOOST_VERSION >= 107000)
+      tcp_connection::sptr new_connection = tcp_connection::make(d_io_service, d_rxbuf.size(), d_tcp_no_delay);
+#else
       tcp_connection::sptr new_connection = tcp_connection::make(d_acceptor_tcp->get_io_service(), d_rxbuf.size(), d_tcp_no_delay);
+#endif
 
       d_acceptor_tcp->async_accept(new_connection->socket(),
         boost::bind(&socket_pdu_impl::handle_tcp_accept, this,
